server/spec/post_bind_bonus_pool_spec.rb

```diff
@@ -409,4 +467,127 @@ def ent_derived_test(pool, expected_quantity)
     @cp.get_entitlement(ent.id)['quantity'].should == 10
   end
 
+  it 'should allow limited consumption of Bonus pools for unlimited master pool created with non-zero virt_limit' do
+    guest_uuid =  random_string('guest')
+    guest_facts = {
+      "virt.is_guest"=>"true",
+      "virt.uuid"=>guest_uuid
+    }
+    guest = @cp.register('guest.bind.com',:system, guest_uuid, guest_facts, 'admin', @owner_key, [], [])
+    guest_cp = Candlepin.new(nil, nil, guest['idCert']['cert'], guest['idCert']['key'])
+
+    # Consume bonus pool (bonus pool quantity == virt_limit == 4)
+    guest_cp.consume_pool(@unlimited_product_bonus_pool['id'], {:quantity => 1})
+    @cp.refresh_pools(@owner_key)
+    bonus_pool = @cp.get_pool(@unlimited_product_bonus_pool['id'])
+    expect(bonus_pool['consumed']).to eq(1)
+
+    # master pool quantity remains unchanged
+    master_pool = @cp.get_pool(@unlimited_product_master_pool['id'])
+    expect(master_pool['quantity']).to eq(-1)
+
+    ent = guest_cp.consume_pool(@unlimited_product_bonus_pool['id'], {:quantity => 3})
+    @cp.refresh_pools(@owner_key)
+
+    expect(@cp.get_pool(@unlimited_product_bonus_pool['id'])['consumed']).to eq(4)
+    # bonus pool is exhausted, no more consumption allowed
+    expect {
+      guest_cp.consume_pool(@unlimited_product_bonus_pool['id'], {:quantity => 4})
+    }.to raise_error(RestClient::Forbidden)
+
+    # unbind & refresh
+    guest_cp.unbind_entitlement(ent[0].id)
+    @cp.refresh_pools(@owner_key)
+    pool = @cp.get_pool(@unlimited_product_bonus_pool['id'])
+
+    expect(pool['consumed']).to eq(1)
+
+    # master pool quantity remains unchanged
+    master_pool = @cp.get_pool(@unlimited_product_master_pool['id'])
+    expect(master_pool['quantity']).to eq(-1)
+  end
+
+  it 'should allow unlimited consumption of master pool regardless of virt_limit' do
+    # unlimited_prod is created with -100 quantity & virt_limit of 4
+    master_pool = @cp.get_pool(@unlimited_product_master_pool['id'])
+    expect(master_pool['quantity']).to eq(-1)
+
+    bonus_pool = @cp.get_pool(@unlimited_product_bonus_pool['id'])
+    expect(bonus_pool['quantity']).to eq(4)
```

Comment:
The premise of this test is correct, but this assertion (and the following ones for the same pool) are wrong, since all bonus pools should always be unlimited when the master pool is unlimited

Reply:
Fixed.